vehicle/src/Vehicle/Data/Builtin/Standard/Type.hs

```diff
@@ -188,10 +191,29 @@ restrictStandardDeclType declSort (ident, p) typ = do
         RestrictedNetwork -> ValidNetworkType
 
   let expr = BuiltinExpr p (TypeClass tc) [explicit typ]
-  let origin = InstanceTypeRestrictionOrigin $ TypeRestrictionOrigin env (ident, provenanceOf typ) declSort typ
+  let origin = InstanceTypeRestrictionOrigin $ TypeRestrictionOrigin env (ident, provenanceOf typ) (Left declSort) typ
   _ <- createFreshInstanceConstraint False mempty p origin Irrelevant expr
   return typ
 
+restrictStandardRecordType ::
+  forall m.
+  (MonadTypeChecker Builtin m) =>
+  DeclProvenance ->
+  Type Builtin ->
+  [RecordField (Type Builtin)] ->
+  m (Type Builtin)
+restrictStandardRecordType (_ident, _p) typ [] = return typ
```

Comment:
Can we use a \case to pattern match on the last argument to avoid rewriting out the name `restrictStandardRecordType` more than necessary?